easyDiffractionApp/Logic/PyQmlProxy.py

```diff
@@ -556,14 +587,31 @@ def _onStructureParametersChanged(self):
     @Slot(str)
     def addSampleFromCif(self, cif_url):
         cif_path = generalizePath(cif_url)
+        if borg.stack.enabled:
+            borg.stack.beginMacro(f'Loaded cif: {cif_path}')
         self._sample.phases = Phases.from_cif_file(cif_path)
+        if borg.stack.enabled:
+            borg.stack.endMacro()
+            # if len(self._sample.phases) < 2:
+            #     # We have problems with removing the only phase.....
+            #     borg.stack.pop()
         self.phaseAdded.emit()
+        # self.undoRedoChanged.emit()
 
     @Slot()
     def addDefaultPhase(self):
         print("+ addDefaultPhase")
+        if borg.stack.enabled:
+            borg.stack.beginMacro('Loaded default phase')
```

Comment:
You are already creating a local accessor for `borg.stack.canUndo()` - can `borg.stack.enabled` be also "localised"?